server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt

```diff
@@ -20,13 +20,17 @@ import suwayomi.tachidesk.manga.model.dataclass.PagedMangaListDataClass
 import suwayomi.tachidesk.manga.model.dataclass.toGenreList
 import suwayomi.tachidesk.manga.model.table.MangaStatus
 import suwayomi.tachidesk.manga.model.table.MangaTable
+import java.lang.IllegalArgumentException
 
 object MangaList {
     fun proxyThumbnailUrl(mangaId: Int): String {
         return "/api/v1/manga/$mangaId/thumbnail"
     }
 
     suspend fun getMangaList(sourceId: Long, pageNum: Int = 1, popular: Boolean): PagedMangaListDataClass {
+        if (pageNum == 0) {
+            throw IllegalArgumentException("Page 0 is an invalid index")
+        }
```

Comment:
I think we should use `require() {}` since its a Kotlin util meant for argument verification

Reply:
Not any of the current clients do it, but its possible if someone starts sending random numbers to the api endpoint

Reply:
Understood, updating...